### PR TITLE
Move SyntaxCollections from gyb to codegen

### PR DESCRIPTION
**Pull request by kimdv**
Part of #1158

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci please test

File: CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift [hunk 1]
```diff
@@ -56,7 +56,7 @@ struct GenerateSwiftSyntax: ParsableCommand {
       TemplateSpec(sourceFile: typeAttributeFile, module: swiftParserDir, filename: "TypeAttribute.swift"),
       TemplateSpec(sourceFile: miscFile, module: swiftSyntaxDir, filename: "Misc.swift"),
       TemplateSpec(sourceFile: syntaxBaseNodesFile, module: swiftSyntaxDir, filename: "SyntaxBaseNodes.swift"),
-      TemplateSpec(sourceFile: syntaxBaseNodesFile, module: swiftSyntaxDir, filename: "SyntaxBaseNodes.swift"),
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift:59]: Why did you remove `syntaxBaseNodesFile`?

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxCollectionsFile.swift [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+                  }
+                }
+              }
+            }
+          }
+          
+          InitializerDeclSyntax("init(_ data: SyntaxData) { self.init(Syntax(data))! }")
+          
+          for choiceName in node.collectionElementChoices ?? [] {
+            if let choiceNode = SYNTAX_NODE_MAP[choiceName] {
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxCollectionsFile.swift:79]: If `choiceName` is not in `SYNTAX_NODE_MAP`, that seems like a pretty serious issue to me. Should we trap instead? I.e. do `choiceNode = SYNTAX_NODE_MAP[choiceName]!`

(same above and below)

File: CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift (hunk 1 above)
Comment (kimdv) [reply CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift:59]: It was there twice. The line above

**Comment by kimdv** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]
As always: Thank you for those changes!

Can we hold back merging this until https://github.com/apple/swift-syntax/pull/1193 is merged? I think the two will conflict and that PR is fixing a more serious bug.

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test
